### DISCUSSION TITLE
Python requirements not installed

### DISCUSSION THREAD

**Discussion by RyanInsolencee** [intelowlproject/IntelOwl#1898] [Q&A]
Hi guys

Ive followed the step by step installation guide but im stuck on starting up IntelOwl after running the [initialize.sh](https://initialize.sh/) script to verify installed dependencies it tells me i can run IntelOwl now after i go into my venv, but when i try to run the [start.py](https://start.py/) i get the error that i dont have the python requirements installed. upon checking on the documents and the requirements, ive verified that i do have them installed. PLEASE HELP I DONT UNDERSTAND AHHHHHHHHHH

**Comment by 0ssigeno** [comment]
Hey.
The script to check the python requirements is kinda straightforward:
```python3
try:
    from dotenv import load_dotenv
    from git import Repo
except ImportError:
    print(
        "you must install the Python requirements."
        " See: https://intelowl.readthedocs.io/en/latest/Installation.html"
    )
    sys.exit(2)
```

   This means that, inside your venv, you do not have these 2 packages installed.

Execute `inizialize.sh` inside your venv (or simply install these 2 packages) should solve the issue.

**Reply by RyanInsolencee** [reply]
i'll try and see what happens thanks alot

**Reply by RyanInsolencee** [reply]
well surprisingly i still get the same error, after running the initialize.sh it says that im ready to go and that all python dependencies are installed

**Reply by 0ssigeno** [reply]
Inside the `venv`, open a python console and run
```python3
    from dotenv import load_dotenv
    from git import Repo
```
 Are you able to do so? Or do you have an `importError` ?
 If this works, `initialize.sh` is not installing the packages at the `venv` level: try to manually install them

**Reply by RyanInsolencee** [reply]
after running the commands in the console i dont get an error, when i try to run the start.py command again it still shows the same issue. im doing this on a ubuntu server 22.04

**Reply by 0ssigeno** [reply]
Mmmmh very strange and I'm not able to reproduce it on my side unfortunately, so I'm going blind with the debug.

Just for testing, can you install these two packages at user level (instead of venv level) and even execute the `start.py` at user level?
I still think that it has something to do with "packages are installed at venv level, and you are executing the script at user level" or viceversa

**Reply by RyanInsolencee** [reply]
i'll give it a try and get back to you

**Reply by RyanInsolencee** [reply]
nope same issue after installing the packages , recreating the venv. FYI this is my 4th VM im doing this on i just keep getting stuck here
i run this on a fresh ubuntu 22.04 server VM

**Reply by 0ssigeno** [reply]
Wait, the `recreating venv` part means that you are still using the venv?

**Reply by 0ssigeno** [reply]
Can you try to follow the guide without using venv at all ? I have absolutely no  idea on why you should not be able to import two packages, i'm lost

**Reply by RyanInsolencee** [reply]
no im at user level, the initialize.sh at the end when everything was done said to recreate the venv which ive done so before

**Reply by RyanInsolencee** [reply]
im as equally lost as you are, even my boss has no idea why this is happening

**Reply by RyanInsolencee** [reply]
After playing around with the start.py( removing the try at the top) and running the script this pops up.
![Screenshot 2023-09-13 175204](https://github.com/intelowlproject/IntelOwl/assets/143370624/cb004a7e-4b25-4c8a-b706-dc57da638ade)

**Reply by RyanInsolencee** [reply]
which again ive verified that dotenv does exist

**Reply by 0ssigeno** [reply]
Another question: do you have multiple python versions in your VM? Maybe you are installing the packages with a version (because pip is connected to, idk, python3.8) and you are executing the venv/start.py with a different version(3.9 for example)?

**Reply by 0ssigeno** [reply]
To check that you can verify that `pip --version` and `python3 --version` refers to the same python version

**Reply by RyanInsolencee** [reply]
both are 3.10
can you list down the all the packages i need or things that is needed to run intelowl and i will check on my end if its correct, unless its too much trouble. NGL been trying this for a week and still getting the same results

**Reply by RyanInsolencee** [reply]
can we get in a discord call and then i can share with you whats going on and show you as well? @0ssigeno

**Reply by 0ssigeno** [reply]
Ye, let me setup a server that we do not actually have one atm for IntelOwl, and could actually be useful for troubleshooting

**Reply by 0ssigeno** [reply]
https://discord.gg/Y83ZyCFh

**Reply by 0ssigeno** [reply]
Unfortunately I can't be online on discord now. You can add screenshot video whatever you want on the `troubleshooting` chan and I will be back to you once I have the time to do so

**Reply by RyanInsolencee** [reply]
just let me know whenever you are free to hop in a quick call. much appreciated

**Comment by mlodic** [comment]
just to hijack this conversation without providing additional help, I have just added an [issue](https://github.com/intelowlproject/IntelOwl/issues/1899) to rewrite the python start.py script in bash so there will be less problems in dependencies

**Comment by chavez243ca** [comment]
Exact same issue here, initialize script either within venv or outside declares all is well, however start.py quickly fails with the error:
```
 $sudo python3 start.py prod up
you must install the Python requirements. See: https://intelowl.readthedocs.io/en/latest/Installation.html
```

**Comment by fgibertoni** [comment]
In some environment you also need the packages from your packet manager in order to comply with those requirements.
Installing `python3-dotenv` and `python3-git` packages should solve the issue.
